### ISSUE TITLE
Reduce data consumption / implement filter usage

### ISSUE THREAD

**Issue by krombel** [vector-im/riot-android#1369]
Currently Riot it the app with most data consumption on my device.
For this I have following idea concerning a "bandwith-friendly-mode":
- reduce traffic (especially on mobile data) to the absolute minimum (e.g. no presence, typing, ...)
- catch up filtered-out data (e.g. presence) when on WiFi (or disabling this option) as sometimes I would like to see the data

**Comment by jcgruenhage** [comment]
Implementing this makes it a lot easier to save battery on the F-Droid build. Once filtering works for this, implementing a "nothing that won't notify the user" filter for background polling shouldn't be hard.

**Comment by ylecollen** [comment]
The presence updates is the feature which uses more data

**Comment by ylecollen** [comment]
https://github.com/matrix-org/matrix-android-sdk/compare/develop...krombel:filter_testing

**Comment by ylecollen** [comment]
@krombel 
Some events are not defined in your filter
- the m.call.* ones
- m.receipt
- m.new_device
- m.room_key
- m.fully_read

**Comment by krombel** [comment]
@ylecollen You are right, i missed `m.call.*`

But:
`m.new_device` and `m.room_key` are part of the `["to_device"]`-part. As I do not limit this, they are part of the response.
Same applies for `m.fully_read` as it is part of `["room"]["account_data"]` but I only set limits to `["room"]["state"]`

To discuss would be `m.receipt` as they are in `["room"]["ephemeral"]` and I disabled those events. So either
- we allow the type `m.receipt` in ephemeral (not allowing type `m.typing` should have the same effect) or
- we only get this type of events on Wifi so let this be synced the same way as `["presence"]`

---

For reference I paste here my suggestion of a filter to be used in the data-saving mode:
```
{
	"room": {
		"state": {
			"types": ["m.room.*", "m.call.*"]
		},
		"ephemeral": {
			"not_types": ["*"]
		}
	},
	"presence": {
		"not_types": ["*"]
	}
}
```
